src/reader.rs

```diff
@@ -4630,6 +4619,49 @@
         zelf.exit_loop_requested |= exit_current_script;
         zelf.parser.libdata_mut().exit_current_script = false;
     }
+
+    /// Execute prompt commands if needed.
+    fn update_prompt(&mut self) {
+        // Clear existing prompts.
+        self.left_prompt_buff.clear();
+        self.right_prompt_buff.clear();
+
+        let exec_left = !self.conf.left_prompt_cmd.is_empty();
+        let exec_right = !self.conf.right_prompt_cmd.is_empty()
+            && (self.conf.right_prompt_cmd != RIGHT_PROMPT_FUNCTION_NAME || {
+                // Load fish_prompt.fish to ensure we don't miss fish_right_prompt if it is defined there.
+                function::load(LEFT_PROMPT_FUNCTION_NAME, self.parser);
+                function::exists(&self.conf.right_prompt_cmd, self.parser)
+            });
+
+        if exec_left || exec_right {
+            self.exec_prompt(exec_left, exec_right);
+        }
+    }
+
+    /// Execute prompt commands if fish in transient mode, passing `--final-rendering` to them.
+    /// Returns whether any commands were executed.
+    fn update_final_prompt(&mut self) -> bool {
+        if !self.conf.transient_prompt {
+            return false;
+        }
+
+        let l_len = self.conf.left_prompt_cmd.len();
+        let r_len = self.conf.right_prompt_cmd.len();
+
+        let exec_left = l_len != 0 && function::exists(&self.conf.left_prompt_cmd, self.parser);
+        let exec_right = r_len != 0 && function::exists(&self.conf.right_prompt_cmd, self.parser);
```

Comment:
this makes me wonder if there is a benefit to executing the prompt if `exec_left` and `exec_left` are both false.
I guess it means we'll execute the mode prompt. Seems fine - less surprising than returning early here I guess.